lib/posthog/client.rb

```diff
@@ -342,10 +346,31 @@ def shutdown
 
     private
 
+    # before_send should run immediately before the event is sent to the queue.
+    # @param [Object] action The event to be sent to PostHog
+    # @return [null, Object, nil] The processed event or nil if the event should not be sent
+    def process_before_send(action)
+      return if action.nil? || action.empty?
+      return action unless @before_send
+
+      processed_action = @before_send.call(action)
+
+      if processed_action.nil?
+        logger.warn("Event #{action[:event]} was rejected in beforeSend function")
+      elsif processed_action.empty?
+        logger.warn("Event #{action[:event]} has no properties after beforeSend function, this is likely an error")
+      end
+
+      processed_action
```

Comment:
Maybe we need a begin/rescue here to avoid problems with the `@before_send` call failing. We can then return `nil` in that case

Reply:
wasn't sure whether to use original or drop on error...

Reply:
went to check JS SDK and that would explode back at the user 🫠

Reply:
It's less bad on JS because `capture` is usually called on useEffects and stuff, but it's much worse in the server where we'll just crash their entire process